src/core/inductive.ml

```diff
@@ -0,0 +1,257 @@
+  | Prod _       -> fatal None "Prod - Not yet implemented"
+  | Abst _       -> fatal None "Abst - Not yet implemented"
+  | LLet _       -> fatal None "LLet - Not yet implemented"
+  | Type         -> fatal None "TYPE - Not yet implemented"(*Pos.none P_Type*)
+  | Kind         -> fatal None "[Kind] not possible"
+  | Wild         -> fatal None "[Wild] not possible"
+  | TRef _       -> fatal None "[TRef] not possible"
+  | Meta _       -> fatal None "[Meta] not possible"
+  | Patt _       -> fatal None "[Patt] not possible"
+  | TEnv _       -> fatal None "[TEnv] not possible"
+
+(** Some constructors to create a p_term without position *)
+
+let _P_Iden : qident -> bool -> p_term = fun ident b ->
+  Pos.none (P_Iden(ident, b))
+
+let _P_Patt : strloc option -> p_term array -> p_term = fun s ts ->
+  Pos.none (P_Patt(s, Some ts))
+
+let _P_Appl : p_term -> p_term -> p_term = fun t1 t2 ->
+  Pos.none (P_Appl(t1, t2))
+
+(** [create_patt t] creates a pattern $[t] without position *)
+let create_patt : strloc -> p_term = fun t -> _P_Patt (Some t) [||]
+
+(*
+let create_arg : int -> p_term list = fun len ->
+  let rec aux : int -> p_term list -> p_term list = fun i acc ->
+    if i = 0 then
+      acc
+    else
+      let head =
+        Pos.none (P_Patt (Some (Pos.none ("a" ^ (string_of_int i))), None))
+      in
+      aux (i-1) (head::acc)
+  in
+  aux len []*)
+
+(** [rec_rules ind_sym rec_sym cons_list] returns the p_rules linking with an
+    induction principle [rec_sym] of the inductive type [ind_sym] (with its
+    constructors [cons_list]). *)
+let rec_rules : sym -> sym -> sym list -> p_rule list =
+  fun ind_sym rec_sym cons_list ->
+
+  (* STEP 0: Define some tools which will be useful *)
+  let app   : p_term -> p_term list -> p_term = List.fold_left _P_Appl in
+  let fapp  : p_term list -> p_term -> p_term = fun ts f -> app f ts in
+(*  let app2  : sym -> sym list -> p_term = fun t q ->
+    let p_patt = create_patt (Pos.none ("pi" ^ t.sym_name)) in
+    List.fold_left (fun q -> _P_Appl q p_patt)
+  in
+  let fapp2 : p_term -> sym list -> p_term = fun p ts -> app2 p ts  in*)
+
+
+  (* STEP 1: Create the common head of the rules *)
+
+  (*let p_iden = _P_Iden (Pos.none ([], sym_ind.sym_name)) true in
+  let p_patt = create_patt (Pos.none "p")                     in
+  let head =
+    List.map (fun c -> create_patt (Pos.none ("pi" ^ c.sym_name))) cons_list
+  in
+  let common_head = fapp head (_P_Appl p_iden p_patt) in*)
+  let p_iden = _P_Iden (Pos.none ([], rec_sym.sym_name)) true in
+  let p_patt = create_patt (Pos.none "p") in
+  let head   = _P_Appl p_iden p_patt in
+  let arg : sym list -> p_term = fun l  ->
+    let rec aux : sym list -> p_term -> p_term = fun l acc ->
+      match l with
+      | []   -> acc
+      | t::q ->
+          let p_patt = create_patt (Pos.none ("p" ^ t.sym_name)) in
+          aux q (_P_Appl acc p_patt)
+    in
+    aux l head
+  in
+  let common_head = arg cons_list in
```

Comment:
This can be simplified as follows:
```
let rec_iden = P.iden rec_sym.sym_name in
let p = P.patt0 "p" in
let head = P.app rec_iden (p :: List.mapi (fun i _ -> P.patt0 (sprintf "p%i" i)))
```

Reply:
You could temporarily use
```
ad = P.app rec_iden (p :: List.map (fun cons_sym -> P.patt0 ("p" ^ cons_sym.sym_name)))
```
for debugging. Replace List.map by List.rev_map if needed.